src/TConsole.cpp

```diff
@@ -712,15 +710,35 @@ void TConsole::closeEvent(QCloseEvent* event)
         }
     }
 
-    if (mType & (SubConsole|UserWindow|Buffer)) {
+    if (mType & (SubConsole|Buffer)) {
         if (mudlet::self()->isGoingDown() || mpHost->isClosingDown()) {
-            std::string key = objectName().toLatin1().data();
-            TConsole* pC = mpHost->mpConsole;
-            if (pC->mSubConsoleMap.find(key) != pC->mSubConsoleMap.end()) {
+            QString key = objectName();
+            auto pC = mpHost->mpConsole->mSubConsoleMap.take(key);
+            if (pC) {
                 mUpperPane->close();
                 mLowerPane->close();
+            }
 
-                pC->mSubConsoleMap.erase(key);
+            event->accept();
+            return;
+        } else {
```

Comment:
Why add the hiding of these and not userwindows?

Reply:
They are - they are treated exactly the same currently (except that i also check for a valid `TDocketWidget` pointer), but I split the code to be a separate block for user windows because I suspect/anticipate that the methodology for them is actually different as hiding the `TDockWidget` that contains the `TConsole` will already have hidden them - and because deleting/closing a QDockWidget automatically hides itself before it deletes IIRC.